examples/impute/plot_multiple_imputation.py

```diff
@@ -0,0 +1,457 @@
+multiple imputation: a method where every missing value is imputed multiple
+times. The procedure results in multiple datasets where the observed data is
+identical in every dataset, but the imputed data is different.
+All desired steps after imputation are performed on every dataset, such as
+standardization, feature engineering and model fitting.
+
+One final model is obtained by combining the estimates of each model with
+Rubin's pooling rules. These rules assume that the parameters of interest are
+normally distributed which is the case with, for example, estimates of the mean
+and regression coefficients. Other parameters, such as correlation
+coefficients need transformation to suit the assumption of normality.
+If it is not possible to approximate a normal distribution, it is better to use
+robust summary measures such as medians or ranges instead of using Rubin's
+pooling rules. This applies to an estimate like explained variance.
+
+In sum, Rubin's pooling rules are as follows. The overall point estimate after
+multiple imputation (denoted by Qbar) is the average of all the m point
+estimates. The variance of the overall point estimate is a combination of
+so-called within imputation variance (Ubar) and between imputation
+variance (B). Ubar is the average of the m variances of the m point estimates.
+Both Qbar and Ubar are corrected with a factor 1 / m to account for sampling
+variance. The between imputation variance (B) is the sum of the squared
+difference between Qbar and the m point estimates, corrected with a factor
+1 / (m – 1). Then, the total variance (T) of the MI overall point estimate is
+Ubar + B + B/m.
+
+In this Example we will show how to use the `sklearn.impute.IterativeImputer`
+to perform multiple imputation. In Example 1 we show the effect of Rubin’s
+pooling rules on the variance of regression estimates. Due to the between
+imputation variance, the standard errors of all regression coefficients are
+larger with multiple imputation than with single imputation.
+
+In Example 2 we show how to set up a prediction model using multiple
+imputation. We combine the predictions from multiple imputations and show that
+this results in better regression performance.
+
+# Authors: Rianne Schouten <r.m.schouten@uu.nl>
+           Sergey Feldman <sergeyfeldman@gmail.com>
+# License: BSD 3 clause
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import stats
+
+from sklearn.datasets import load_boston
+from sklearn.linear_model import RidgeCV
+from sklearn.preprocessing import StandardScaler
+from sklearn.model_selection import train_test_split
+from sklearn.impute import IterativeImputer
+from sklearn.metrics import mean_squared_error as mse
+
+rng = np.random.RandomState(0)
+
+N_ITER = 20  # number of iterations in IterativeImputer
+N_SIM = 3  # number of simulations in Example 2
+
+
+# Start by defining a basic amputation function
+def ampute(X, missing_rate=0.75, mech="MCAR"):
```

Comment:
Please add a brief docstring at least describing what ampute means and MCAR vs MNAR

Reply:
rewrite this function:
```suggestion
def ampute(X, missing_rate=0.75, mech="MCAR"):
	X_out = X.copy()
	n_drop_per_feature = int(missing_rate * X.shape[0])
	for x in np.transpose(X_out):
	    # insert missing values for each feature
	    if mech == 'MCAR':
	        prob = None
	    else:
	        weights = scipy.special.expit(x - x.mean())
	        prob = weights / weights.sum()
	    drop_idx = np.random.choice(X.shape[0], p=prob, replace=False,
	                                size=n_drop_per_feature)
	    x[drop_idx] = np.nan
	return X_out
```

Reply:
We don't even use MNCAR in this example because it's so computationally expensive. I'll simplify and remove references to MNCAR.

Reply:
Well, I guess it might be useful to have if it's only a few lines of code as you rewrote it.

Reply:
@jnothman Is there a plan to have an ``ampute`` function in ``sklearn.datasets``?

Reply:
I would think about `strategy` instead of `mech`?